### DISCUSSION TITLE
Add RUAdlist Bitblock

### DISCUSSION THREAD

**Discussion by electrifying** [uBlockOrigin/uBOL-home#251] [Ideas]
since a recent update to ublock lite added an optional subscription to ruadlist counters could you also add the bitblock subscription, yes it contains some of the rules from the fanboy's list but now the number of unique rules in the subscription exceeds 40% according to the author dimisa. So can we see this subscription in future ublock updates?
https://easylist-downloads.adblockplus.org/bitblock.txt

**Comment by gorhill** [comment]
What's the purpose of BitBlock?

There is a [limit to the maximum number of static rulesets](https://developer.chrome.com/docs/extensions/reference/api/declarativeNetRequest#property-MAX_NUMBER_OF_STATIC_RULESETS) which can be packed in the extension, so arguments to add more lists will have to be solid.

**Reply by mapx-** [reply]
**BitBlock**

The filter list contains almost the entire Fanboy's/EasyList Annoyance List (and Fanboy-Social included there, respectively), with the exception of private rules for English-language sites and EasyList Cookie. Due to the fact that Fanboy filters do not always work correctly in the RuNet, BitBlock contains rules to neutralize false hiding and blocking that occurs when Fanboy filters work.

https://forums.lanik.us/viewtopic.php?f=102&t=22512

**Reply by dportvine** [reply]
@dimisa-RUAdList

**Reply by dimisa-RUAdList** [reply]
The main task of BitBlock is to make Fanboy filters work correctly for regions targeted by the RU AdList. To achieve the current result, BitBlock has been developed for more than 10 years and as far as I can judge, it is a fairly popular filter. The current uBOL interface does not have information about limits, so I do not know how critical this is.

**Reply by gorhill** [reply]
> to make Fanboy filters work correctly

Is it a complement or a replacement of Fanboy Annoyances?

**Reply by dimisa-RUAdList** [reply]
Of the 24 thousand rules, the number unique to BitBlock is 11 thousand and they are in addition to the 13 thousand general rules of Fanboy: https://github.com/easylist/ruadlist/blob/master/bitblock.txt

It is assumed that the BitBlock can work independently.

**Reply by dportvine** [reply]
If anyone is interested in my opinion as a uBO user. I connected BitBlock back in 2017. Since then, I see how this filter is actively filled and it happens quickly. You don't need to wait a week or a month. If I start actively reporting Social Widgets and Annoyances on UA ​​sites to EasyList, it will take a lot of time. Even if the rules are updated once a week when a new version of uBOL is released, BitBlock will be much more effective than similar filters. 

As an example, my report through ABP, this has already been decided https://github.com/easylist/ruadlist/commit/8f2ce11c79038a88ae962605087303e97ed66e25

<details>
<summary>Screen</summary>

![2024-12-05_223849](https://github.com/user-attachments/assets/13733c4e-832a-495d-b0ab-57ad94aa0f03)
</details>

Other sites where a Fanboy subscription won't hide social widgets.

`https://www.eurointegration.com.ua/news/2024/12/4/7199893/`
`https://www.rbc.ua/rus/news/ssha-rozpovili-udari-tsilyah-siriyi-vidpovid-1733289128.html`
`https://news.novyny.live/u-ssha-stalos-kilka-potuzhnikh-zemletrusiv-nablizhaietsia-tsunami-217602.html`

<details>
<summary>Screen</summary>

![2024-12-05_223849](https://github.com/user-attachments/assets/a2a8c6d9-ceba-4589-bd19-522e82084c02)
![2024-12-05_224549](https://github.com/user-attachments/assets/e5424899-d373-4ac8-90ad-ea980ed20829)
![2024-12-05_224607](https://github.com/user-attachments/assets/f50de8e6-9b83-49d2-881b-35515bb8ae7a)
</details>

And many other sites...

>It is assumed that the BitBlock can work independently.

If there is a limits, maybe this can be a solution.

**Reply by electrifying** [reply]
yes i agree, if the rule limit allows it, it would be nice to see bitblock in subscriptions

**Reply by belnet** [reply]
I join the suggestion of dimisa, bitblock subscription is extremely necessary so as not to add it manually every time.

**Reply by aprilman** [reply]
also join the suggestion of dimisa

**Reply by gorhill** [reply]
@dimisa-RUAdList What if:

- duplicates from Fanboy lists are removed from Bitblock
- then merged with Counter and the list is renamed "RU AdList: Counters + Bitblock"

The idea is that:

- the number of available static rulesets is limited in uBOL
- to ensure the overhead of content scripts to inject does not grow unduly
- to ensure the package size does not grow unduly

**Reply by dimisa-RUAdList** [reply]
And how much does BitBlock in its current form not fit within the limits?

The described method of changes, which involves deleting Fanboy duplicates, will lead to the filter ceasing to do its intended work, since the general fanboy rules are the skeleton of this list. Conventionally, the BitBlock work scheme can be described as follows:
1. The general rules fanboy_notifications, fanboy_social, fanboy_annoyance are triggered, a total of 13 thousand rules.
2. Regional (unique to BitBlock) rules are triggered, which fix the breakdowns caused by the triggering of the general fanboy rules, and block/hide those requests and elements that did not fall under the general fanboy rules. There are about 11 thousand such regional rules.

**Comment by Mombasazzz** [comment]
It's a great idea to add BitBlock!!!
I always use this subscription too.

**Comment by Yuki2718** [comment]
There's also a limit to the maximum number of lists. How many are left for that currently?

**Reply by MasterKia** [reply]
Enabled rulesets: 50
Total rulesets: 100
![image](https://github.com/user-attachments/assets/6f1675b2-1fba-4cb1-a460-e56b9f4482f2)


uBOL has 47 rulesets in total:
![image](https://github.com/user-attachments/assets/5357e8fa-fbc0-4a8e-8e60-d3ce26d4c3de)

**Reply by electrifying** [reply]
@gorhill Hello, can you give an answer whether it will be possible to add bitblock or will it greatly affect the limit? What is it at the moment. Thanks for answer

**Reply by gorhill** [reply]
No plan to add Bitblock for the moment. I tried to propose a compromise to make it work but this was ruled out. I know there is space to add more lists but I can't keep adding special purpose lists for one single region because this opens the door for endless requests to do the same for all other regions in which case we run out of available rulesets. I will be very stringent on keeping adding regional rulesets -- once we add these, it will be difficult to remove them in the future.
